### PR TITLE
docs: simplify getting started guide to use native concurrency and local build

### PR DESCRIPTION
**Pull request by kpenfound**
Simplifying the getting started guide by removing the errgroup concurrency since the DAG can solve it's own concurrency. Also build the project from host.workdir instead of a git repo.

Please review the revised guide and edit 🙏 

We also want this guide to reference the programming model. I can update that in this PR or we can make that change in another PR

### CODE REVIEW

**Comment by talios** [comment]
One thing I was experimenting with, was calling:

```go
 errGroup.SetLimit(concurrentJobs)
```

to limit the number of concurrent builds happening - mostly this was attempting to mitigate memory issues with concurrent java builds etc.

Is there a more idiomatic dagger way of limiting the concurrency?

**Review by aluzzardi — APPROVED** [review]
Code LGTM

**Comment by aluzzardi** [comment]
> to limit the number of concurrent builds happening - mostly this was attempting to mitigate memory issues with concurrent java builds etc.

@talios As far as I know, there is a setting to limit the concurrency however it's 1) global and 2) it's not exposed (/cc @sipsma @vito not sure if that's accurate -- referring to `MaxParallelism`).

You can however control the concurrency natively in Go.

- Adding something like `ExitCode(ctx)` at the end of an `Exec()` will make it synchronous
- You can then decide whether you want to keep everything synchronous, or using errgroup  with `SetLimit(concurrentJobs)` as you were saying

File: docs/current/sdk/go/959738-get-started.md [hunk 1]
```diff
@@ -156,11 +152,10 @@ This revision of the Go CI tool adds another layer to the build matrix, this tim
 Try the Go CI tool by executing the commands below:
 
 ```shell
-go build
-./multibuild https://github.com/kpenfound/greetings-api.git
+go run main.go
 ```
 
-The Go CI tool clones the Git repository, builds the application for each OS/architecture/version combination and writes the results to the host. You will see the build process run eight times, once for each combination.
+The Go CI tool builds the application for each OS/architecture/version combination and writes the results to the host. You will see the build process run eight times, once for each combination. Note that the each build is happening concurrently, because each build in the DAG do not depend on eachother.
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/sdk/go/959738-get-started.md:158]: Minor suggestion, approving so you can merge

**Review by vikram-dagger — APPROVED** [review]

**Pull request merged**